CrashReporter/mainwindow.cpp

```diff
@@ -9,78 +9,171 @@ MainWindow::MainWindow(QWidget *parent) :
 {
-    QString s3Host = "s3.amazonaws.com";
+    qDebug() << _process->readAllStandardOutput() << endl;
+}
 
-    QS3::S3 s3(s3Host);
-    QS3::Bucket* bucket = s3.bucket(this->bucketName, this->accessKey, this->secretKey);
 
-    const QDateTime nowTime = QDateTime::currentDateTime(); 
-	const QString timestamp = nowTime.toString(QLatin1String("yyyyMMdd-hhmmsszzz"));
-	QString AutosavePath;
-    QString userMessage;
-    if (this->windowState == 1) {
-        userMessage = "email: " + ui->lineEdit->text() + "\nmessage:" + ui->plainTextEdit->toPlainText() + "\n" + "log: " + this->logInformation;
-        AutosavePath = timestamp + "_Crash.log";
-    } else {
-        userMessage = "email: " + ui->lineEdit->text() + "\nmessage:" + ui->plainTextEdit->toPlainText() + "\n";
-        AutosavePath = timestamp + "_Feedback.log";
-    }
-    QByteArray data(userMessage.toUtf8());
-    bucket->upload(AutosavePath, data);
+void MainWindow::processError(QProcess::ProcessError perr)
+{
 
-    QEventLoop loop;
-    connect(bucket, SIGNAL(finished()), this, SLOT(uploadToS3Done()));
-    loop.exec();
+    qDebug() << "process Error " << perr  << endl;
+    qDebug() << _process->program() << " " <<  _process->arguments();
 
+    QCoreApplication::quit();
 
 }
 
-void MainWindow::on_pushButton_2_clicked()
+void MainWindow::readError()
 {
+    qDebug() << _process->readAllStandardError() << endl;
+
+}
+
+void MainWindow::started()
+{
+    qDebug() << "process started " << _process->program() << " " << _process->processId() << endl;
+}
+
+void MainWindow::finished(int exitCode)
+{
+    // check the exit status and exit code.
+    // if the node program failed to upload the logs, inform the user about it
+    qDebug() << endl << " process finished : " << exitCode << endl;
     QCoreApplication::quit();
+
+}
+
+void MainWindow::processLogs()
+{
+    for(const QString& path: _logsPath) {
+        QFile fptr(path);
+
+        if(fptr.open(QIODevice::ReadOnly)) {
+            std::string log = fptr.readAll().toStdString();
+            std::string::size_type n;
+
+            while(log.find(' ') != std::string::npos) {
+               n = log.find(' ');
+               log.replace(n, 1, "__");
+            }
+
+            while(log.find('\n') != std::string::npos) {
+               n = log.find('\n');
+               log.replace(n, 1, "###");
+            }
+
+            _logs += QString(log.c_str());
+            fptr.close();
+        }
+    }
+}
+
+void MainWindow::uploadLogs()
+{
+    processLogs();
+
+    QStringList args;
+    args.append(_script);
+    args.append(_logs);
+
+    _process->setArguments(args);
+
+    if(_process->state() == QProcess::Running)
+        _process->kill();
+
+    _process->start();
+}
+
+void MainWindow::startElMaven()
+{
+    if (this->windowState == 1) {
+        QProcess * myProcess = new QProcess();
+        myProcess->start(this->restartApplicationPath);
+    }
+
 }
 
 void MainWindow::uploadToS3Done() {
```

Comment:
@rish9511 Lets remove this function?

Reply:
@shubhra-agrawal 
Let's do this later ?